multiqc/modules/sambamba/markdup.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python
+
+from __future__ import print_function
+from collections import OrderedDict
+import os
+import logging
+import re
+
+from multiqc import config
+from multiqc.plots import bargraph
+
+log = logging.getLogger(__name__)
+
+class SambambaMarkdupMixin:
+
+    """ Find and parse Sambamba Markdup output log files and calculate duplication rate """
+
+    def parse_sambamba_markdup(self):
+
+        # Clean sample name from 'markdup_sample_1' to 'sample_1'
+        # Find and load sambamba logs to markdup_data_dict.
+        # Regex for key phrases and calculate duplicate rate.
+        # Give user warning if redundant samples are found.
+
+        self.markdup_data_dict = dict()
+
+        for f in self.find_log_files("sambamba/markdup"):
+
+            sample_name = self.markdup_clean_s_name(f["s_name"])
+            self.markdup_data_dict[sample_name] = self.get_markdup_stats(f["f"])
+
+            self.add_data_source(f)
+
+            if f["s_name"] in self.markdup_data_dict:
+                log.debug("Duplicate sample name found! Overwriting: {}".format(f["s_name"]))
+
+        if len(self.markdup_data_dict) == 0:
+            raise UserWarning
+
+        # Write parsed files to a file
+        self.markdup_write_data_to_file()
+
+        # Add sambamba markdup to general statistics table
+        self.markdup_general_stats_table()
+
+        # Add sambamba markdup bargraph section
+        self.markdup_section()
+
+        # return the length of log files parsed
+        return len(self.markdup_data_dict)
+
+    def markdup_clean_s_name(self, s_name):
+
+        """
+        Clean up sample name from 'markdup_sample_1' to 'sample_1'.
+        Is there a better way to do this?
+        """
+
+        return s_name.replace("markdup_", "")

```

Comment:
Not really - there is `fn_clean_exts` and `fn_clean_trim` in the config but they both work with suffixes. I don't think that there is any equivalent for prefixes.. 🤔 

I think that your approach is fine, but I would (a) not use a custom function just for a `.replace()` call 😅 and (b) you need to use `self.clean_s_name()` - see https://multiqc.info/docs/#custom-sample-names

I'll leave a suggested change above to give an example of what I mean.